iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -254,6 +263,35 @@ public void commitAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable
     }
   }
 
+  @Override
+  public void rollbackAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable, EnvironmentContext context)
+      throws MetaException {
+    if (Boolean.valueOf(context.getProperties().getOrDefault(MIGRATE_HIVE_TO_ICEBERG, "false"))) {
+      LOG.debug("Initiating rollback for table {} at location {}",
+          hmsTable.getTableName(), hmsTable.getSd().getLocation());
+      context.getProperties().put(INITIALIZE_ROLLBACK_MIGRATION, "true");
+      this.catalogProperties = getCatalogProperties(hmsTable);
+      try {
+        this.icebergTable = Catalogs.loadTable(conf, catalogProperties);
+      } catch (NoSuchTableException nte) {
+        // iceberg table was not yet created, no need to delete the metadata dir separately
+        return;
+      }
+
+      // we want to keep the data files but get rid of the metadata directory
+      hmsTable.getParameters().put(InputFormatConfig.EXTERNAL_TABLE_PURGE, "FALSE");
```

Comment:
Maybe I'm confused, but I thought this was to be removed?

Reply:
I'm confused too :). This shouldn't be there.